lib/bridge-client/upload-state.js

```diff
@@ -56,14 +55,17 @@ UploadState.DEFAULTS = {
 /**
  * Unlinks the referenced tmp files
  */
-UploadState.prototype.cleanup = function() {
+UploadState.prototype.cleanup = function(cb) {
   this.killed = true;
 
-  this.cleanQueue.forEach(function(tmpFilePath) {
-    if (utils.existsSync(tmpFilePath)) {
-      rimraf.sync(tmpFilePath);
-    }
-  });
+  async.each(this.cleanQueue, function(options, cb) {
+    options.store.exists(options.key, function(err, exists) {
```

Comment:
Do we set `options.key` anywhere? Looks like we are leaving the temporary files behind.